### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by pieterbos**
Updated the README and some documentation of classes, and added a BuiltinReferenceModels class to easily obtain model metadata. Added an archie-all dependency that depends on the entire library.

Also removed some unused files.

Fixes https://github.com/openEHR/archie/issues/11 and https://github.com/openEHR/archie/issues/9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,50 +88,59 @@ CAttribute attribute = archetype.getDefinition()
...
+The BMM models are a file containing metadata in a form defined by the OpenEHR specifications. 
+The reflection based metadata contains ModelInfoLookup classes. They are derived from an implementation of a reference model. Note that the ModelInfoLookup classes are only added if you depended on them. If you depended on archie-all, you're all set.
 
-```java
-String serialized = ADLArchetypeSerializer.serialize(archetype);
-```
+### Operational templates
 
-### Flattener and Operational Template creation
+You can create operational templates of your archetypes. Think of operational templates as something you generate from an archetype or template, to more easily work with. While a source archetype or even a flattened archetype can still contain 'use_node' for example, these have all been replaced. And all use_archetype c_archetype_roots have been replaced with the structure of the arhetype. See the OpenEHR OPT2 specification for what it does.
```
Comment (steijgeler) [inline README.md]: This part could use some rephrasing

File: README.md [hunk 2]
```diff
@@ -9,52 +9,33 @@ It uses the ANTLR adl-grammar written by Thomas Beale at https://github.com/open
...
-    - RMQueryContext
-    - ... probably more
-- RMQueryContext now needs a JAXBContext for the RM to function. You can use JAXBUtil.getArchieJaxbContext() for the openehr-rm 
-- The ArchetypeRepository interface now has a getAll() method. It's likely this interface will be renamed or removed in the future in favor of what is now the FullArchetypeRepository
-- OverridingArchetypeRepository.addArchetype has been renamed to addExtraArchetype
-- ADLPArser.withConstrainsImposer has been removed. You can still do new ADLPArser(new RMConstraintsImposer()) if you want, or use it manually. 
-- all datetime parsers are now located in com.nedap.archie.datetime.DateTimeParsers
-- CObject is now an abstract class, as it should be according to specs
-- ModelInfoLookup now also returns computed attributes, with isComputed() returning true. This can break lots of RM processing tools!
+## Dependency
```
Comment (steijgeler) [inline README.md]: Maybe add an TOC to the README? It's quite long

File: README.md [hunk 3]
```diff
@@ -364,16 +361,82 @@ And many more things. Currently supported is comparison and arithmetic operators
...
+for(String resource:resourceNames) {
+    try(InputStream odin = TestUtil.class.getResourceAsStream(resource)){
+        models.getAomProfiles().add(odin);
+    } catch (IOException e) {
+        throw new RuntimeException(e);
+    }
+}        
+```
+
+This instantiates a MetaModels class that has both the archie reference model implementation, the BMM models and AOM profiles. The BMM models and AOM profileswill be used for the flattener and archetype validator, the other models for tools that work on reference models.
```
Comment (steijgeler) [inline README.md]: profileswill

**Comment by pieterbos** [comment]
rebasing no longer possible due to git magic on other branches i did. opening a new PR

**Pull request closed without merging**